### PR TITLE
James 3226 Automated site build and deploy 

### PR DESCRIPTION
**Pull request by ieugen**
Primary goals:
* The goal of this branch is to have a version of the James documentation online. 
* It's the goal of this branch to have automated builds for the documentation.
* Have a solution for handling the old website content.

Secondary goals:
* Clenaup and remove old junk files - but not old documentation

Non Goals:
* Have the production website online - we can track that once we have a working, documented, automated process. 

This branch should be merged when we have the following:

Proposal:
* [] A documented build process for the documentation
* [x] An automated version of that process
* [x] A working build on builds.apache.org with github integration

You can submit PR's against this branch for specific items. 
We should update the proposal as it is necessary. 

Incorporates:
* https://issues.apache.org/jira/browse/JAMES-3226
* https://issues.apache.org/jira/browse/JAMES-3294

### CODE REVIEW

**Comment by ieugen** [comment]
I've pushed a version to build a website using gradle. 
This should work on local and builds.a.o . 
See the README.adoc file. 
Use `./gradlew clean build` to get a sample website.

**Comment by dleangen** [comment]
@ieugen cool, thank you!! I will check tomorrow.

**Comment by ieugen** [comment]
I've added a Jenkinsfile and setup the automated build https://builds.apache.org/view/J/view/ApacheJamesProjects/job/ApacheJames-Website/ .

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        // If this build didn't fail, but there were failing tests, send an email to the list.
+        unstable {
+            echo "Unstable "
+        }
+
+        // Send an email, if the last build was not successful and this one is.
+        success {
+            echo "Success "
+            script {
+                if ((currentBuild.previousBuild != null) && (currentBuild.previousBuild.result != 'SUCCESS')) {
```
Comment (aduprat) [inline Jenkinsfile:97]: :+1: 
It's great to don't have mail on every builds.

**Comment by ieugen** [comment]
Local build using gradle and the James website.
(Antora default UI for now) 
Captură de ecran de la 2020-07-06 19-24-13

**Comment by dleangen** [comment]
Nice job!!! Man you are dedicated!

I can confirm that I can build the site.

Questions:
 - what do we do about the branding? (I think we need to build our own branding zip file)
 - is the removal of jsieve just temporary? (I think we should incorporate *all* the James-related docs)
 - is it intentional to point to HEAD? (Antora is built to have several versions simultaneously running side by side)

**Comment by ieugen** [comment]
I think branding can be a local gradle project that we assemble with distribution plugin and use it in antora (I hope it works with file Uri).

Sieve was not working so I commented out. 
Please contributeand make things better. 
I will focus on automation until it's done. 
Let me know if we need something else. 
I'm currently working on publishing the staging site so we can see it remotely.

**Comment by dleangen** [comment]
Ok, I think I understand what you are trying to do, and I appreciate it very much.

Once the automation is working, and now that I have commit access, it will be much easier for me to update the site. I will update the structure etc. as needed, don't worry about that.

**Comment by ieugen** [comment]
I've pushed manually to the staging site: https://james.staged.apache.org/ . 
I've copied the files from build/site and committed them. 
I'll see how we can automate this steps.

**Comment by dleangen** [comment]
Again, very nice!! Let me know if there is anything specific you would like. Otherwise I am going to continue to concentrate my efforts of the content, since now I am getting unblocked a bit.

**Comment by ieugen** [comment]
We can publish on build: 
https://builds.apache.org/view/J/view/ApacheJamesProjects/job/ApacheJames-Website/view/change-requests/job/PR-1/2/console

File: doc-sites/build.gradle [hunk 2]
```diff
@@ -9,14 +11,34 @@ node {
...
+
   inputs.files('package.json', 'package-lock.json', 'antora-playbook.yml')
-  inputs.dir(fileTree("node_modules").exclude(".cache"))
-  outputs.dir('build/site')
+  inputs.dir(fileTree('node_modules').exclude('.cache'))
+  outputs.dir(gitPublish.repoDir)
 
   command = 'antora'
-  args = ['antora-playbook.yml']
+  args = ['antora-playbook.yml', '--to-dir', siteOutputDir ]
```
Comment (chibenwa) [inline doc-sites/build.gradle]: This antore playbook is uncomitted?

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+== How to build the website
+
+[source,bash]
+----
+    # To build the website run
+    ./gradlew clean build
+    # The website is located here
+    cd doc-sites/build/site
+----
```
Comment (chibenwa) [inline README.adoc:44]: Can we add how to serve locally the documentation website to the README ?

File: doc-sites/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,1876 @@
+{
```
Comment (chibenwa) [inline doc-sites/package-lock.json:1]: Do we want to commit this lock file?

**Review by chibenwa — APPROVED** [review]
Overall a huge :+1: , thanks for coming up with this!

File: doc-sites/package-lock.json (hunk 4 above)
Comment (ieugen) [reply doc-sites/package-lock.json:1]: Yes. Lock files mean the versions are fixed and you don't get surprises when you npm install

File: README.adoc (hunk 3 above)
Comment (ieugen) [reply README.adoc:44]: Just open the files. Antora URL work without a http server. 
It's a feature they advertise.

**Comment by ieugen** [comment]
[JAMES-3226] Closing this as we have moved the website build code to `live` branch.

**Pull request closed without merging**